### ISSUE TITLE
Orderbooks doesn't get all the offers 

### ISSUE THREAD

**Issue by tuloski** [ripple/ripple-lib#371]
Hello everybody.
I'm subscribing to some orderbooks with the 'model' event and as far as I understood the orderbook object should gather all the actual offers on the orderbook (snapshot) and keep it updated as soon as new offers arrive.
But it seems to me that not all the offers are collected. I noticed this because I subscribed to some of the most liquid markets (as XRP:USD.snapswap, XRP:USD.bitstamp, all BTC:XRP and so on) and only a total of 300 offers (more or less) are collected but only the XRP:USD.bitstamp orderbook has more than 300 offers. Moreover the offers are really slow to fill. Do I see such small amount of offers because the orderbook filters the unfunded offers?

And another question about orderbooks: how can I know if the offers on the orderbooks object are updated with the last ledger? I saw there is a private _syncronized var. Is there a way to know if the orderbook is actually syncronized with the latest closed ledger? Because the oderbook is slowly filled offer-by-offer and it seems there is no way to understand when it is up to date.

thanks

**Comment by mwshortt** [comment]
Hello! I'm working on reproducing this issue, could you share the code that's causing this problem with me?

**Comment by tuloski** [comment]
Hello,
this is the code (more or less):

function requestAllBooks(markets, remote) {
    for (var i = 0; i < markets.length; i++) {
        for (var j = 0; j < markets.length; j++) {
            if (i!=j) {
                var book = remote.createOrderBook({
                    currency_gets: markets[i].currency,
                    issuer_gets: markets[i].issuer,
                    currency_pays: markets[j].currency,
                    issuer_pays:markets[j].issuer
                });
                book.on("model", handle_book);
                book.on('offer_funds_changed', handle_book_funds);
            }
        }
    }
}

where 'markets' is an array of markets and in particular:

if (!configuration.TEST_MARKETS){   //real one
    markets.push(new market('USD', BITSTAMP, 0, 0, 0.2));
    markets.push(new market('USD', SNAPSWAP, 0, 0, 0.2));
    markets.push(new market('BTC', BITSTAMP, 0, 0, 0.2));
    markets.push(new market('BTC', SNAPSWAP, 0, 0, 0.2));
    markets.push(new market('CNY', RIPPLECN, 0, 0, 0.2));
    markets.push(new market('JPY', RIPPLETRJAPAN, 0, 0, 1));
    markets.push(new market('JPY', TOKYOJPY, 0, 0, 0));
    markets.push(new market('EUR', SNAPSWAP, 0, 0, 0.2));
    markets.push(new market('CNY', RIPPLEFOX, 0, 0, 0));
    markets.push(new market('EUR', GATEHUB, 0, 0, 0.2));
    markets.push(new market('USD', GATEHUB, 0, 0, 0.2));
    markets.push(new market('XRP', null, 0, 0, 0));
}

so basically there is a lot of subscriptions (12x12 = 144 orderbooks) and it is strange that the screenshot is more or less 200-300 offers.
I'm subscribed to a public RL server.

And what about the 'synchronized'?

**Comment by clark800** [comment]
The order book only shows 300 orders because that is the maximum number that can be obtained from public rippled servers. If you run your own rippled server then you can get more results. Also, there is a ticket which will allow getting all the orders from rippled, but it is not currently prioritized, so it may not be addressed anytime soon: https://ripplelabs.atlassian.net/browse/RIPD-817.

Regarding the question about "synchronized", I've opened a ticket to look into it: https://ripplelabs.atlassian.net/browse/RLJS-414
